RecoLocalTracker/SiStripZeroSuppression/src/SiStripRawProcessingFactory.cc

```diff
@@ -12,24 +13,26 @@
 #include "RecoLocalTracker/SiStripZeroSuppression/interface/FastLinearCMNSubtractor.h"
 #include "RecoLocalTracker/SiStripZeroSuppression/interface/TT6CMNSubtractor.h"
 
-std::unique_ptr<SiStripRawProcessingAlgorithms> SiStripRawProcessingFactory::create(const edm::ParameterSet& conf) {
+std::unique_ptr<SiStripRawProcessingAlgorithms> SiStripRawProcessingFactory::create(const edm::ParameterSet& conf,
+                                                                                    edm::ConsumesCollector&& iC) {
   return std::unique_ptr<SiStripRawProcessingAlgorithms>(
-      new SiStripRawProcessingAlgorithms(create_SubtractorPed(conf),
-                                         create_SubtractorCMN(conf),
+      new SiStripRawProcessingAlgorithms(std::move(iC),
```

Comment:
Mostly for my own understanding, what is in C++ the effect of repeatedly `std::move` the same rValue reference to the ConsumesCollector? I couldn't find other similar examples.

Reply:
The first move leaves the moved-from object into an unspecified but valid state (and e.g. with `std::unique_ptr` the moved-from state is specified, it is null pointer). In general moving from the same object is an error.

For `edm::ConsumesCollector` specifically moving doesn't add any value. It is essentially a pointer, and can be (nowadays) copied cheaply. I'd recommend to just pass `edm::ConsumesCollector` by value, unless there is a desire to pass it through some intermediate class with forward declaration only, in which case by (regular/lvalue) reference would be the way to go (unless that function should bind also a temporary `ConsumesCollector`, in which case rvalue reference would be needed).

Reply:
@pieterdavid @makortel : then, what is the suggestion/plan for this part of the code?

Reply:
I followed existing code where I saw it mostly being taken by rvalue reference, but if it doesn't matter and copies are cheap I'll change it to pass it by value or lvalue reference, as @makortel suggested, in this kind of cases.

Reply:
This should be fixed with the latest changes (passing the consumescollector by value between the zero-suppression factory methods and constructors); I also made initialising the token optional again in the few cases where it was only used depending on some flag (that should be a bit more robust in the case they are run without the full database, even if I don't think this is done in practice)